aibolit/patterns/instanceof/instance_of.py

```diff
@@ -15,16 +15,32 @@ def __traverse_node(self, node):
         List of code lines
         """
         lines = []
-        for _, node_elem in node.filter(javalang.tree.BinaryOperation):
-            if node_elem.operator == 'instanceof' and node_elem.operandl.position is not None:
-                code_line = node_elem.operandl.position.line or node_elem.operandr.position.line
-                lines.append(code_line)
-        for _, node_elem in node.filter(javalang.tree.MethodInvocation):
-            if node_elem.member == 'isInstance':
-                lines.append(node_elem.position.line)
-
+        depth = 0
+        cur_line = 0
+        bin_depth = -10
+        m_depth = -10
+        for _, destination, edge_type in dfs_labeled_edges(node.tree, node.root):
+            cur_node = node.tree.nodes[destination]
+            if edge_type == 'forward':
+                cur_node = node.tree.nodes[destination]
+                node_type = cur_node['type']
+                if 'source_code_line' in cur_node:
+                    cur_line = cur_node['source_code_line']
+                if node_type == ASTNodeType.BINARY_OPERATION:
+                    bin_depth = depth
+                if node_type == ASTNodeType.STRING and cur_node['string'] == 'instanceof' and depth - bin_depth == 1:
+                    lines.append(cur_line)
+                    bin_depth = -10
+                if node_type == ASTNodeType.METHOD_INVOCATION:
```

Comment:
Checking for binary operations and method invocations could be split into two different function.

Reply:
Are you sure that it makes sense?